wp-content/themes/oceanwp-child/functions.php

```diff
@@ -125,8 +125,116 @@ function custom_repeatable_meta_box_save($post_id) {
 	}
 }
 
+function initiative_meta_boxes() {
+	add_action('admin_init', 'ggrc_add_initiative_meta_boxes', 2);
+
+	function ggrc_add_initiative_meta_boxes() {
+		add_meta_box( 'ggrc_initiative_updates', 'Initiative Updates', 'Repeat_meta_box_display', 'initiative', 'normal', 'default');
+	}
+
+	function updates_row_template($updates, $field = null) {
+		?>
+			<tr>
+				<td> 
+					<input required type="text" placeholder="Title" name="UpdateTitle[]" <?php if ($field) echo 'value="' . $field['UpdateTitle'] . '"' ?>/>
+				</td>
+				<td> 
+					<input required type="date" placeholder="Date" name="UpdateDate[]" <?php if ($field) echo 'value="' . $field['UpdateDate'] . '"' ?>/>
+				</td>
+				<td> 
+                    <textarea rows="3" placeholder="Initiative Updates" name="Update[]"><?php if ($field) echo $field['Update'] ?></textarea>
+					
+				</td>
+				<td>
+					<a class="button cmb-remove-row-button remove-row <?php if ($field === null) echo 'button-disabled'; ?>" href="#">Remove</a>
+				</td>
+			</tr>
+		<?php
+	}
+
+	function Repeat_meta_box_display() {
+		
+		global $post;
+
+		$ggrc_initiative_updates = get_terms( array(
+			//'taxonomy' => 'initiative_updates',
+			'hide_empty' => false,
+		) );
+		$ggrc_initiative = get_post_meta($post->ID, 'initiative', true);
+		
+		 wp_nonce_field( 'ggrc_repeat_meta_box_nonce', 'ggrc_repeat_meta_box_nonce' );
+		?>
+		<script type="text/javascript">
+		jQuery(document).ready(function( $ ){
+			const tbody = document.querySelector("#repeatable-fieldset-one tbody");
+
+			$( '#add-row' ).on('click', function() {
+				const rowWrapper = document.createElement('tbody');
+				rowWrapper.innerHTML = `<?php updates_row_template($ggrc_initiative_updates); ?>`;
+				const row = rowWrapper.children[0];
+				const lastChild = tbody.children[tbody.children.length - 1];
+				tbody.insertBefore( row, lastChild );
+				return false;
+			});
+	
+			$( '.remove-row' ).on('click', function() {
+				$(this).parents('tr').remove();
+				return false;
+			});
+		});
+		</script>
+		<table id="repeatable-fieldset-one" width="100%">
+		<tbody>
+			<?php
+			if ( $ggrc_initiative ) :
+				foreach ( $ggrc_initiative as $field ) {
+					updates_row_template($ggrc_initiative_updates, $field);
+				}
+			else :
+				// show a blank one
+				updates_row_template($ggrc_initiative_updates);
+			endif; ?>
+		</tbody>
+		</table>
+		<p><a id="add-row" class="button" href="#">Add another</a></p>
+		<?php
+	}
```

Comment:
Extract `Repeat_meta_box_display` for reuse (also used in news - and might be needed elsewhere. 
Not for this PR.

Reply:
I tried extracting it and separating but didn't exactly work right. I tried it for initiative updates but it adds the field on the all initiative lists page. I'm not too sure if I made any mistakes or what but code seemed right. See the function below : 
[generaladdrepeatablemetabox.txt](https://github.com/dev-a-better-world/ggrc_website/files/8131251/generaladdrepeatablemetabox.txt)

Reply:
As I said - `not for this PR`.